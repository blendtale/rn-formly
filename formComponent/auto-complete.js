import React, {useRef, useState} from 'react'
import {
  View, 
  Text, 
  TextInput, 
  StyleSheet, 
  FlatList, 
  Dimensions, 
  TouchableOpacity,
  ActivityIndicator
} from 'react-native'
import PropTypes from 'prop-types'


const DEFAULT_ERROR_MESSAGE = 'Error While Loading the data'

const styles = StyleSheet.create({
  defaultTextInputStyle: {
    width: 'auto',
    marginTop: 50,
    padding: 5,
    justifyContent: 'center',
    borderBottomWidth: 1, 
    width: Dimensions.get('window').width * 0.8,
    fontSize: 24,
    marginBottom: 20
  }, 
  modelOpenViewMain: {
    zIndex: 1,
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height,
    ...StyleSheet.absoluteFillObject,
    backgroundColor:'rgba(0,0,0,0.7)',
    flex: 1,
    display: 'flex', 
    flexDirection: 'column', 
    alignItems: 'center'
  },
  modelOpenViewSub: {
    backgroundColor:'rgba(255,255,255, 0.9)',
    display: 'flex', 
    flexDirection: 'column',
    alignItems: 'center',
    height: Dimensions.get('window').height*0.4,
    width: Dimensions.get('window').width * 0.9,
    paddingBottom: 10,
    marginTop: Dimensions.get('window').height * 0.1
  }, 
  flatlistIndividualComponent: {
    height: 30, 
    marginTop: 20,
    paddingRight: 10
  }, 
  flatListIndividualComponentText: {
    fontSize: 21
  }, 
  loadingView: {
    display: 'flex', 
    flexDirection: 'column', 
    justifyContent: 'center', 
    alignItems: 'center',
    height: '50%'
  }
})

const AutoComplete = (props) => {
    const {
    upsideEmit,
    defaultColor,
    value,
    asyncFunction,
    textInputStyle,
    loaderRequired,
    listTextStyle,
    listViewStyle,
    ...rest
  } = props
  let parentViewRef = null
  const [layoutProperties, setLayoutProperties] = useState(null)
  const [modalOpen, setModelOpen] = useState(false)
  const [suggestions, setSuggestion] = useState([])
  const [loading, setLoadingStatus] = useState(false)

  
  const onFocus = async (text) => {
    if (parentViewRef.measure && !layoutProperties) {
      parentViewRef.measure( (fx, fy, width, height, px, py) => {
        setLayoutProperties({
          width: width, 
          height: height, 
          offsetX: px,
          offsetY: py
        })
      })
    }
    //To check if values have data or not
    const valueArr = Object.values(value)
    if (valueArr.length > 0 && value.title)  {
      const suggestions = await asyncFunction(value.title)
      setSuggestion(suggestions)
    }
  }

  const onchangeHandler = async (text) => {
    try {
    if (loaderRequired) setLoadingStatus(true)
    if(layoutProperties && !modalOpen) setModelOpen(true)
    const emitObj = {
      title: text
    }
    const suggestions = await asyncFunction(text)
    setSuggestion(suggestions)
    upsideEmit(emitObj)
    } catch (error) {
      if(modalOpen) setModelOpen(false)
      if (error.message)  upsideEmit(null, error.message)
      else upsideEmit(null, DEFAULT_ERROR_MESSAGE)
    }
    if (loaderRequired) setLoadingStatus(false)
  }

  const onSelection = (selectedObj) => {
    if(modalOpen) setModelOpen(false)
    upsideEmit(selectedObj)
  }

  if (!modalOpen) {
      return (
        <View>
            <TextInput
            {...rest}
            ref={component => parentViewRef = component}
            style={[{color: defaultColor, borderColor: defaultColor}, styles.defaultTextInputStyle, textInputStyle]}
            onChangeText={(text) => onchangeHandler(text)}
            onFocus={onFocus}
            value={value.title}
          />
        </View>
      )
  } else {
    return (
      <View style={[styles.modelOpenViewMain, {marginTop: -layoutProperties.offsetY + 83.5}]}>
       <View needsOffscreenAlphaCompositing style={styles.modelOpenViewSub}>
        <TextInput
          {...rest}
          ref={component => parentViewRef = component}
          style={[{color: defaultColor, borderColor: defaultColor}, styles.defaultTextInputStyle, textInputStyle]}
          onChangeText={(text) => onchangeHandler(text)}
          onFocus={onFocus}
          value={value.title}
          autoFocus={true}
          />
          { loading ? (
            <View style={styles.loadingView}>
              <ActivityIndicator size="large" color={defaultColor} />
            </View>
           ): (
            <FlatList 
            data={suggestions}
            renderItem={({ item }) => (
              <TouchableOpacity onPress={() => onSelection(item)} style={[styles.flatlistIndividualComponent, listViewStyle]}>
                <Text style={[styles.flatListIndividualComponentText, listTextStyle]}> {item.title}</Text>
            </TouchableOpacity>
            )} 
            keyExtractor={(item) => item.id.toString()}/>
           )
            }
          </View>
        </View>
    )
  }
}

AutoComplete.propTypes = {
  upsideEmit: PropTypes.func.isRequired,
  defaultColor: PropTypes.string,
  value: PropTypes.object,
  textInputStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  asyncFunction: PropTypes.func.isRequired,
  loaderRequired: PropTypes.bool, 
  listTextStyle: PropTypes.object, 
  listViewStyle: PropTypes.object
}

export default AutoComplete