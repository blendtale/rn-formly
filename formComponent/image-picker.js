import React, { useState, useEffect} from 'react';
import ImagePicker from 'react-native-image-crop-picker';
import {View, Text, StyleSheet, TouchableOpacity, Dimensions, Image} from 'react-native'
import PropTypes from 'prop-types'


const styles = StyleSheet.create({
  mainImageContainer: {
    width: Dimensions.get('window').width * 0.8,
    maxWidth:  Dimensions.get('window').width * 0.8,
    marginTop: 20,
    display: 'flex', 
    alignItems: 'center'
  },
  // Single Image Styling
  imageView: {
    width: Dimensions.get('window').width * 0.6,
    height: Dimensions.get('window').height * 0.3,
    backgroundColor: '#F5F5F5'
  }, 
  circleView: {
    borderRadius: Math.round((Dimensions.get('window').width*0.6) + (Dimensions.get('window').height*0.3)) / 2,
    display: 'flex',
    alignItems: 'flex-end', 
    justifyContent: 'flex-end'
  }, 
  TextPlusSign: {
    fontSize: 72
  }, 
  multipleImageContainerSubChild: {
    display: 'flex', 
    flexDirection: 'row',
    flexWrap: 'wrap',
    alignItems: 'center'
  },
  multipleImageContainerSingleChild: {
    width: 85,
    height: 100,
    marginLeft: 6,
    marginRight: 6,
    marginTop: 5,
    marginBottom: 5,
    backgroundColor: '#F5F5F5',
    display: 'flex',
    alignItems: 'flex-end', 
    justifyContent: 'flex-end'
  },
  multipleImagePlus: {
    fontSize: 32,
  }
  // Multiple Images Styling 
})

export const SingleImageSelector = (props) => {
  const {
    upsideEmit,  
    cropWidth, 
    cropHeight, 
    imageComponentStyle, 
    containerShape, 
    defaultColor,
  } = props
  const objectCheck = props.value ? Object.keys(props.value) : null
  const value = objectCheck && objectCheck.length > 0 ? props.value : null
  const selectImage = async () => {
    try {
      const imageDocRecieved = await ImagePicker.openPicker({
          width: cropWidth,
          height: cropHeight,
          cropping: true
        })
        upsideEmit(imageDocRecieved)  
      } catch (error) {
      }
  }

  if (!value) {
    return (
      <View style={styles.mainImageContainer}>
        <TouchableOpacity onPress={selectImage}
        style={[styles.imageView, styles.circleView, imageComponentStyle]}> 
          <Text style={[styles.TextPlusSign, {color: defaultColor}]}> + </Text>
        </TouchableOpacity>
      </View>
    )
  } else {
    return (
      <View style={styles.mainImageContainer}>
      <TouchableOpacity onPress={selectImage}
      style={[styles.imageView, styles.circleView, imageComponentStyle]}> 
      <Image source={{uri: value.path}} style={[styles.imageView, styles.circleView, imageComponentStyle]} />
      </TouchableOpacity>
      </View>
    )
  }
}


SingleImageSelector.propTypes = {
  upsideEmit: PropTypes.func.isRequired,
  cropHeight: PropTypes.number,
  cropWidth: PropTypes.number, 
  imageComponentStyle: PropTypes.object, 
  defaultColor: PropTypes.string, 
  value: PropTypes.object
}


export const MultipleImageSelector = (props) => {
  //Filling objects with null array 
  const {
    upsideEmit,  
    cropWidth, 
    cropHeight, 
    imageComponentStyle, 
    containerShape, 
    defaultColor,
    maximumNoOfImages
  } = props

  const value = props.value && Array.isArray(props.value) ? props.value :  Array(maximumNoOfImages ? maximumNoOfImages: 9).fill(null)
  const selectImage = async (index) => {
    try {
      const imageDocRecieved = await ImagePicker.openPicker({
          width: cropWidth,
          height: cropHeight,
          cropping: true
        })
        const currentImageDoc = [...value]
        currentImageDoc[index] = imageDocRecieved
        upsideEmit(currentImageDoc) 
      } catch (error) {
        console.warn(error)
      }
    }

  return (
    <View style={styles.mainImageContainer}>
      <View style={styles.multipleImageContainerSubChild}>
        {value.map((el, index) => {
          if (el) {
            return (
            <TouchableOpacity 
            onPress={() => selectImage(index)}
            key={index} 
            style={[styles.multipleImageContainerSingleChildImage, imageComponentStyle]}>
              <Image source={{uri: el.path}} style={[styles.multipleImageContainerSingleChild, imageComponentStyle]} />
            </TouchableOpacity>
            )
          } else {
            return (
              <TouchableOpacity 
                onPress={() => selectImage(index)}
                key={index} 
                style={[styles.multipleImageContainerSingleChild, imageComponentStyle]}>
              <Text style={[styles.multipleImagePlus, {color:defaultColor}]}>+</Text>
              </TouchableOpacity>
            )
          }
        })}
      </View>
    </View>
  )
}


MultipleImageSelector.propTypes = {
  upsideEmit: PropTypes.func.isRequired,
  cropHeight: PropTypes.number,
  cropWidth: PropTypes.number, 
  imageComponentStyle: PropTypes.object, 
  defaultColor: PropTypes.string, 
  value: PropTypes.array,
  maximumNoOfImages: PropTypes.number
}


