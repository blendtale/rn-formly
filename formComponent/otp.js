import React from 'react'
import {StyleSheet} from 'react-native'
import PropTypes from 'prop-types'
import NextTextInput from 'react-native-next-input'

const styles = StyleSheet.create({
  textInputStyle: {
    textAlign: 'center', 
    marginLeft: 5, 
    marginRight: 5, 
    fontSize: 16,
    borderBottomWidth: 1,
    color: 'black'
  }
})

const Otp = ({
  noOfTextInput,
  upsideEmit,
  value,
  textInputStyle,
  inputTextPlaceHolder
}) => {
  const onChangeHandler = (dateArray, currentValue, refForTheCurrentValue) => {
    const stringifyArray = dateArray.join('')
    if (stringifyArray.length === noOfTextInput) {
      upsideEmit(dateArray)
    }
  }

  return (
   <>
    <NextTextInput
    onChangeValue={onChangeHandler}
    noOfTextInput={noOfTextInput}
    textInputStyle={[styles.textInputStyle, textInputStyle]}
    placeholder={inputTextPlaceHolder}
    value={value}
  />
   </>
  )
}

Otp.propTypes = {
  noOfTextInput: PropTypes.number.isRequired,
  upsideEmit: PropTypes.func.isRequired,
  value: PropTypes.array,
  textInputStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
}


export default Otp
