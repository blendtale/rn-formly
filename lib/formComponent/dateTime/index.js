import * as React from 'react';
import styles from './styles';
import NextTextInput from 'react-native-next-input';
const ERROR_DEFAULT_MESSAGE = 'Invalid Date, Please check the entered date again';
// Used for Date
const dateFormatter = {
    DDMMYYYY: (input) => {
        const formatInput = input.join('');
        const date = new Date(formatInput.replace(/(\d{2})(\d{2})(\d{4})/, '$2-$1-$3'));
        if (date instanceof Date)
            return date;
        else
            throw new Error(ERROR_DEFAULT_MESSAGE);
    },
    MMDDYYYY: (input) => {
        const formatInput = input.join('');
        const date = new Date(formatInput.replace(/(\d{2})(\d{2})(\d{4})/, '$1-$2-$3'));
        if (date instanceof Date)
            return date;
        else
            throw new Error(ERROR_DEFAULT_MESSAGE);
    },
    YYYYMMDD: (input) => {
        const formatInput = input.join('');
        const date = new Date(formatInput.replace(/(\d{4})(\d{2})(\d{2})/, '$3-$2-$1'));
        if (date instanceof Date)
            return date;
        else
            throw new Error(ERROR_DEFAULT_MESSAGE);
    },
};
const DateTyper = ({ dateFormat = 'DDMMYYYY', textInputStyle, upsideEmit, value = [], ...rest }) => {
    if (value.length > 8) {
        console.warn('Value supplied for data Typer might be incorrect');
    }
    // if array, convert to string
    let formatedValue = value && Array.isArray(value) ? value.join('') : value;
    // Creating placeholder for our Next Input component
    const inputTextPlaceHolder = dateFormat.split('');
    // On change Value
    const onChangeHandler = (dateArray, currentValue, refForTheCurrentValue) => {
        // Would filter out value
        const filterArray = dateArray.filter((value) => value);
        if (filterArray.length === 8) {
            //update in state
            const date = dateFormatter[dateFormat](dateArray).toString().split('');
            upsideEmit(date, currentValue, refForTheCurrentValue);
        }
    };
    const textInputMergedStyle = {
        ...styles.textInputStyle,
        textInputStyle,
    };
    return (React.createElement(React.Fragment, null,
        React.createElement(NextTextInput, { onChangeValue: onChangeHandler, noOfTextInput: 8, textInputStyle: textInputMergedStyle, placeholder: inputTextPlaceHolder, value: formatedValue.split('') })));
};
export default DateTyper;
//Formats
