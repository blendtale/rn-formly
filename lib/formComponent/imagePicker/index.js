import * as React from 'react';
import ImagePicker from 'react-native-image-crop-picker';
import { View, Text, TouchableOpacity, Image } from 'react-native';
import styles from './style';
export const SingleImageSelector = ({ upsideEmit, cropWidth, cropHeight, imageComponentStyle, defaultColor, value, }) => {
    const selectImage = async () => {
        try {
            const imageDocRecieved = await ImagePicker.openPicker({
                width: cropWidth,
                height: cropHeight,
                cropping: true,
                multiple: false,
            });
            if (Array.isArray(imageDocRecieved)) {
                throw new Error('Bug in rn-formly library, Expected single image but recieved multiple Image');
            }
            upsideEmit(imageDocRecieved);
        }
        catch (error) {
            throw new Error(error);
        }
    };
    if (!value) {
        return (React.createElement(View, { style: styles.mainImageContainer },
            React.createElement(TouchableOpacity, { onPress: selectImage, style: [styles.imageView, styles.circleView, imageComponentStyle] },
                React.createElement(Text, { style: [styles.TextPlusSign, { color: defaultColor }] },
                    ' ',
                    "+",
                    ' '))));
    }
    else {
        return (React.createElement(View, { style: styles.mainImageContainer },
            React.createElement(TouchableOpacity, { onPress: selectImage, style: [styles.imageView, styles.circleView, imageComponentStyle] },
                React.createElement(Image, { source: { uri: value.path }, style: [styles.imageView, styles.circleView, imageComponentStyle] }))));
    }
};
export const MultipleImageSelector = ({ upsideEmit, cropWidth, cropHeight, imageComponentStyle, defaultColor, maximumNoOfImages, value, }) => {
    //Filling objects with null array
    const imageValue = value && Array.isArray(value)
        ? value
        : Array(maximumNoOfImages ? maximumNoOfImages : 9).fill(null);
    const selectImage = async (index) => {
        try {
            const imageDocRecieved = await ImagePicker.openPicker({
                width: cropWidth,
                height: cropHeight,
                cropping: true,
            });
            const currentImageDoc = [...imageValue];
            currentImageDoc[index] = imageDocRecieved;
            upsideEmit(currentImageDoc);
        }
        catch (error) {
            throw new Error(error);
        }
    };
    return (React.createElement(View, { style: styles.mainImageContainer },
        React.createElement(View, { style: styles.multipleImageContainerSubChild }, imageValue.map((el, index) => {
            if (el) {
                return (React.createElement(TouchableOpacity, { onPress: () => selectImage(index), key: index, style: [
                        styles.multipleImageContainerSingleChild,
                        imageComponentStyle,
                    ] },
                    React.createElement(Image, { source: { uri: el.path }, style: [
                            styles.multipleImageContainerSingleChild,
                            imageComponentStyle,
                        ] })));
            }
            else {
                return (React.createElement(TouchableOpacity, { onPress: () => selectImage(index), key: index, style: [
                        styles.multipleImageContainerSingleChild,
                        imageComponentStyle,
                    ] },
                    React.createElement(Text, { style: [styles.multipleImagePlus, { color: defaultColor }] }, "+")));
            }
        }))));
};
