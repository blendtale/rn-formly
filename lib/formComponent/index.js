export { default as DateTyper } from './dateTime';
export { default as CheckBoxes } from './checkbox';
export { SingleImageSelector, MultipleImageSelector } from './imagePicker';
export { default as Otp } from './otp';
export { InputText } from './text';
export { default as AutoComplete } from './autoComplete';
const something = {
    name: 'CÃ©lio',
    lastName: 'Garcia',
};
const options = { ...something };
const { email, phone, name, lastName } = options;
