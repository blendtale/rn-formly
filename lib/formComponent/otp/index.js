import * as React from 'react';
import NextTextInput from 'react-native-next-input';
import styles from './style';
const Otp = ({ noOfTextInput, upsideEmit, value, textInputStyle, inputTextPlaceHolder, ...rest }) => {
    const onChangeHandler = (dateArray, currentValue, refForTheCurrentValue) => {
        const stringifyArray = dateArray.join('');
        if (stringifyArray.length === noOfTextInput) {
            upsideEmit(dateArray, currentValue, refForTheCurrentValue);
        }
    };
    const textStyles = {
        ...styles.textInputStyle,
        ...textInputStyle,
    };
    return (React.createElement(React.Fragment, null,
        React.createElement(NextTextInput, Object.assign({ onChangeValue: onChangeHandler, noOfTextInput: noOfTextInput, textInputStyle: textStyles, placeholder: inputTextPlaceHolder, value: value }, rest))));
};
export default Otp;
