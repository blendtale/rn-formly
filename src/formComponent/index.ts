export {default as DateTyper} from './dateTime'
export {default as CheckBoxes} from './checkbox'
export {SingleImageSelector, MultipleImageSelector} from './imagePicker'
export {default as Otp} from './otp'
export {InputText} from './text'
export {default as AutoComplete} from './autoComplete'