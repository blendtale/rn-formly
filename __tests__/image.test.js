import React from 'react'
import {SingleImageSelector} from './../formComponent/image-picker'
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });


describe('Check if image component renders', () => {
    test('image component renders', () => {
        const props = {
            upsideEmit: () => {},
        }
      const wrapper = shallow(<SingleImageSelector {...props}/>);
      expect(wrapper).toMatchSnapshot();
    });
  });