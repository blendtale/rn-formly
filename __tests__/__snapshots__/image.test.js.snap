// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if image component renders image component renders 1`] = `ShallowWrapper {}`;
